Sources/MapboxCoreNavigation/NavigationEventsManager.swift

```diff
@@ -202,27 +182,31 @@ open class NavigationEventsManager {
         return eventDictionary
     }
     
-    func navigationFeedbackEvent(type: FeedbackType, description: String?) -> EventDetails? {
-        guard let dataSource = dataSource, let sessionState = sessionState else { return nil }
-
-        var event = ActiveGuidanceEventDetails(dataSource: dataSource, session: sessionState, defaultInterface: usesDefaultUserInterface)
-        event.event = MMEEventTypeNavigationFeedback
+    func navigationFeedbackEvent(type: FeedbackType, description: String?) -> NavigationEventDetails? {
+        var event: NavigationEventDetails?
         
-        event.userId = UIDevice.current.identifierForVendor?.uuidString
-        event.feedbackType = type.description
-        event.description = description
+        if let activeNavigationDataSource = dataSource, let sessionState = sessionState {
+            event = ActiveNavigationEventDetails(dataSource: activeNavigationDataSource,
+                                                 session: sessionState, defaultInterface: usesDefaultUserInterface)
+        } else if let passiveNavigationDataSource = passiveNavigationDataSource {
+            event = PassiveNavigationEventDetails(dataSource: passiveNavigationDataSource)
+        } else {
+            assertionFailure("NavigationEventsManager is unable to create feedbacks without a datasource.")
+        }
         
-        event.screenshot = captureScreen(scaledToFit: 250)?.base64EncodedString()
+        event?.feedbackType = type.description
```

Comment:
We should try to keep this property strongly typed as an enumeration until the last possible moment before serialization as JSON. The code has been pulling out the description like this, but this is an opportunity to introduce a bit more safety using FeedbackType.